docgen/src/guide/Refining_hits.md

```diff
@@ -0,0 +1,552 @@
+
+## Regular setup
+
+First, we'll set up a regular project with create-react-app and react-instantsearch.
+
+```sh
+$ yarn create react-app refining-hits # or npm install -g create-react-app && create-react-app refining-hits
+$ cd refining-hits
+$ yarn add react-instantsearch # or npm install --save react-instantsearch
+```
+
+The index that we will use in this example needs to be set up with: 
+
+```jsx
+index.setSettings({
+  attributesForFaceting: ['searchable(keywords)', 'searchable(owner.name)'],
+});
+```
+
+In this guide we will use the search index we use on the [Yarn](https://yarn.fyi) website. It already has the mandatory `attributesForFaceting ` configured.
+We will get started in `App.js` and set up a regular React InstantSearch app:
+
+```jsx
+import React, { Component } from 'react';
+import {
+  Configure,
+  InstantSearch,
+  SearchBox,
+  Hits,
+} from 'react-instantsearch/dom';
+
+class App extends Component {
+  render() {
+    return (
+      <InstantSearch
+        appId="OFCNCOG2CU"
+        apiKey="f54e21fa3a2a0160595bb058179bfb1e"
+        indexName="npm-search"
+      >
+        <SearchBox />
+        <CurrentRefinements />
+        <Hits />
+        <Configure
+          attributesToRetrieve={[
+            'name',
+            'description',
+            'keywords',
+            'owner.name',
+          ]}
+        />
+      </InstantSearch>
+    );
+  }
+}
+
+export default App;
+```
+
+We will also set up a separate file called `Hit.js` for the component that renders the hits: 
+
+```jsx
+import React from 'react';
+
+const Tags = ({ keywords = [] }) =>
+  <div>
+    {keywords.map((keyword, i) =>
+      <span key={keyword}>
+        {i > 0 && ', '}
+        <button>
+          {keyword}
+        </button>
+      </span>
+    )}
+  </div>;
+
+const Hit = ({ hit: { name, description, keywords, owner } }) =>
+  <article>
+    <h1>
+      <a
+        href={`https://yarn.fyi/${name}`}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        {name}
+      </a>
+    </h1>
+    <p>
+      {description}
+    </p>
+    <p>
+      by{' '}
+      <button>
+        {owner.name}
+      </button>
+    </p>
+    <Tags keywords={keywords} />
+  </article>;
+
+export default Hit;
+```
+
+In this component we took the name, description and owner of a certain package, and render these as hits. To enable this as `hitComponent`, we will change the line in `App.js` that renders the `Hit` component:
+
+```jsx
+import React, { Component } from 'react';
+import {
+  Configure,
+  InstantSearch,
+  SearchBox,
+  Hits,
+} from 'react-instantsearch/dom';
+import Hit from './Hit'
+
+class App extends Component {
+  render() {
+    return (
+      <InstantSearch
+        appId="OFCNCOG2CU"
+        apiKey="f54e21fa3a2a0160595bb058179bfb1e"
+        indexName="npm-search"
+      >
+        <SearchBox />
+        <CurrentRefinements />
+        <Hits hitComponent={({hit}) => <Hit hit={hit} />}/>
```

Comment:
We can't pass an anonymous function to the `hitComponent`. We can't use it as a render prop because, inside `Hits` the function will be used with `createElement`. Then since on every render the reference of the function will change React will unmount & mount the `Hit` component on every render.

I have made a little example of both approach, you can see that on each render (trigger by `forceUpdate`) the each `Hit` component is unmount / mount.

> https://jsfiddle.net/samuel_vllnt/uwpw621o/